src/tiledb/cloud/vcf/query.py

```diff
@@ -0,0 +1,364 @@
+    :param verbose: verbose logging, defaults to False
+    :return: Arrow table containing the query results
+    """
+    import tiledbvcf
+
+    # Adjust inputs
+    if config is None:
+        config = {}
+    if isinstance(attrs, str):
+        attrs = [attrs]
+    if isinstance(regions, str):
+        regions = [regions]
+    if isinstance(samples, str):
+        samples = [samples]
+
+    logger = setup(config, verbose)
+    logger.debug("tiledbvcf=%s", tiledbvcf.version)
+
+    config["rest.use_refactored_array_open"] = True
+
+    # TODO: evaluate TileDB config options
+    # Avoid issue loading tile offsets for a large number of fragments
+    # config["sm.mem.reader.sparse_unordered_with_dups.ratio_array_data"] = 0.4
+    # config["sm.mem.reader.sparse_unordered_with_dups.ratio_coords"] = 0.4
+    # Alternative to above
+    # config["sm.partial_tile_offsets_loading"] = True
+
+    # Define TileDB-VCF config
+    cfg = tiledbvcf.ReadConfig(
+        region_partition=region_partition,
+        sample_partition=sample_partition,
+        memory_budget_mb=memory_budget_mb,
+        tiledb_config=config,
+    )
+
+    if verbose:
+        tiledbvcf.config_logging("debug")
+
+    with Profiler(array_uri=log_uri, id=log_id) as prof:
+        # Open TileDB-VCF Dataset
+        ds = tiledbvcf.Dataset(dataset_uri, cfg=cfg)
+
+        # Issue read query
+        tables = [
+            ds.read_arrow(
+                attrs=attrs,
+                regions=regions,
+                bed_file=bed_file,
+                samples=samples,
+                set_af_filter=af_filter or "",
+            )
+        ]
+
+        # Loop over any incomplete queries
+        while not ds.read_completed():
+            tables.append(ds.continue_read_arrow())
+
+        # Combine any incomplete queries into a single arrow table
+        table = pa.concat_tables(tables)
+
+        prof.write("result", table.num_rows, table.nbytes)
+
+    memory_usage_gb = max_memory_usage() / (1 << 30)
+    logger.debug("Max memory usage: %0.3f GiB", memory_usage_gb)
+    logger.debug("Incomplete queries: %d", len(tables) - 1)
+    logger.debug("Records read: %d", table.num_rows)
+    logger.debug("Arrow table size: %0.3f MiB", table.nbytes / (1 << 20))
+
+    return table
+
+
+def concat_tables_udf(
+    tables: List[pa.Table],
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    log_uri: Optional[str] = None,
+    verbose: bool = False,
+) -> pa.table:
+    """
+    Concatenate a list of Arrow tables.
+
+    :param tables: Arrow tables
+    :param config: config dictionary, defaults to None
+    :param log_uri: log URI for profiling, defaults to None
+    :param verbose: verbose logging, defaults to False
+    :return: concatenated Arrow table
+    """
+
+    logger = setup(config, verbose)
+
+    with Profiler(array_uri=log_uri) as prof:
+        table = pa.concat_tables(
+            [x for x in tables if x is not None and x.num_rows > 0]
```

Comment:
nit: no need for the `[]`s in here; giving it an iterable of tables will work fine